velox/functions/prestosql/window/Rank.cpp

```diff
@@ -41,17 +51,36 @@ class RankFunction : public exec::WindowFunction {
       vector_size_t resultOffset,
       const VectorPtr& result) override {
     int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
-    auto* rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
     auto* rawPeerStarts = peerGroupStarts->as<vector_size_t>();
+    auto rawValues = result->asFlatVector<T>()->mutableRawValues();
+
     for (int i = 0; i < numRows; i++) {
       auto start = rawPeerStarts[i];
       if (start != currentPeerGroupStart_) {
         currentPeerGroupStart_ = start;
-        rank_ += previousPeerCount_;
+        if constexpr (type == RankType::kDenseRank) {
+          if (previousPeerCount_ > 0) {
+            ++rank_;
+          }
+        } else {
+          rank_ += previousPeerCount_;
+        }
         previousPeerCount_ = 0;
       }
 
-      rawValues[resultOffset + i] = rank_;
+      switch (type) {
+        case RankType::kRank:
+        case RankType::kDenseRank: {
+          rawValues[resultOffset + i] = rank_;
```

Comment:
To avoid switching on every row, use if constexpr or add a helper function with 2 specializations.

Reply:
Thanks, used ``if constexpr``.